src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -7130,24 +7144,188 @@ public override IOperation VisitArgument(IArgumentOperation operation, int? capt
+
+                        var badAssignment = new SimpleAssignmentOperation(isRef: false, badTarget, valueCaptureRef,
+                            semanticModel: null, operation.Syntax, valueCaptureRef.Type, constantValue: null, isImplicit: true);
+
+                        initializerBuilder.Add(badAssignment);
+                        continue;
+                    }
+
+                    var propertyReference = (IPropertyReferenceOperation)simpleAssignment.Target;
+
+                    Debug.Assert(propertyReference != null);
+                    Debug.Assert(propertyReference.Arguments.IsEmpty);
+                    Debug.Assert(propertyReference.Instance != null);
+                    Debug.Assert(propertyReference.Instance.Kind == OperationKind.InstanceReference);
+                    Debug.Assert(((IInstanceReferenceOperation)propertyReference.Instance).ReferenceKind == InstanceReferenceKind.ImplicitReceiver);
 
-            IOperation cloned = operation.CloneMethod is null
-                ? MakeInvalidOperation(visitedInstance.Type, visitedInstance)
-                : new InvocationOperation(operation.CloneMethod, visitedInstance,
-                    isVirtual: true, arguments: ImmutableArray<IArgumentOperation>.Empty,
+                    var property = propertyReference.Property;
+                    var assignment = makeAssignment(property, valueCaptureRef, operation);
+
+                    explicitProperties.Add(property.Name, assignment);
+                }
+
+                // Make a sequence for all properties (in order), constructing assignments for the implicitly set properties
+                var type = (INamedTypeSymbol)operation.Type;
+                foreach (IPropertySymbol property in properties)
+                {
+                    if (explicitProperties.TryGetValue(property.Name, out var assignment))
+                    {
+                        initializerBuilder.Add(assignment);
+                    }
+                    else
+                    {
+                        Debug.Assert(oldValueCaptureId >= 0);
+
+                        // `oldInstance`
+                        var oldInstance = new FlowCaptureReferenceOperation(oldValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        // `oldInstance.Property`
+                        var visitedValue = new PropertyReferenceOperation(property, ImmutableArray<IArgumentOperation>.Empty, oldInstance,
+                            semanticModel: null, operation.Syntax, property.Type, isImplicit: true);
+
+                        int extraValueCaptureId = GetNextCaptureId(outerCaptureRegion);
+                        AddStatement(new FlowCaptureOperation(extraValueCaptureId, operation.Syntax, visitedValue));
+
+                        LeaveRegionsUpTo(innerCaptureRegion);
```

Comment:
>LeaveRegionsUpTo(innerCaptureRegion); [](http://example.com/codeflow?start=24&length=37)

It doesn't look like we are visiting anything here. Is there a reason to manipulate with regions?